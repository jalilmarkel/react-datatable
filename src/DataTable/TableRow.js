import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme, css } from 'styled-components';
import TableCell from './TableCell';

const TableRowStyle = styled.tr`
  border-top: 1px solid ${props => props.theme.rows.borderColor};
  ${props => props.striped && css`
      &:nth-child(odd) {
        background-color: ${props.theme.rows.stripedColor};
      }
  `};
  ${props => props.highlightOnHover && css`
      &:hover {
        background-color: ${props.theme.rows.hoverColor};
        transition-duration: 0.15s;
        transition-property: background-color;
      }
  `};
`;

const TableRow = ({ striped, highlightOnHover, children, columns, keyField, row, index, onRowClicked }) => {
  const handleRowClick = e =>
    onRowClicked && onRowClicked(row, index, e);

  return (
    <TableRowStyle
      striped={striped}
      highlightOnHover={highlightOnHover}
      onClick={handleRowClick}
    >
      {children}
      {columns.map(col => (
        <TableCell
          type="cell"
          key={`cell-${col.id}-${row[keyField] || index}`}
          width={col.width}
          column={col}
          row={row}
        />))}
    </TableRowStyle>
  );
};

TableRow.propTypes = {
  striped: PropTypes.bool,
  highlightOnHover: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  columns: PropTypes.array.isRequired,
  keyField: PropTypes.string.isRequired,
  row: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  onRowClicked: PropTypes.func,
};

TableRow.defaultProps = {
  striped: false,
  highlightOnHover: false,
  children: null,
  onRowClicked: null,
};

export default withTheme(TableRow);
