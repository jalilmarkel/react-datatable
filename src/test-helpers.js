import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

