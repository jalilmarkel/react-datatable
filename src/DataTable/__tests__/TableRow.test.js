import 'jest-styled-components';

import React from 'react';
import { shallowWithTheme } from '../../test-helpers';

import TableRow from '../TableRow';

test('component <TableRow /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableRow columns={[]} row={{}} keyField="id" index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableRow striped /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableRow striped columns={[]} row={{}} keyField="id" index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableRow highlightOnHover /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableRow highlightOnHover columns={[]} row={{}} keyField="id" index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableRow /> with children should render correctly', () => {
  const wrapper = shallowWithTheme(<TableRow columns={[]} row={{}} keyField="id" index={0}><td>Some Child Cell</td></TableRow>);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableRow /> with columns should render correctly if a row does not have a keyField', () => {
  const columns = [{
    id: 123,
    name: 'Name',
    selector: 'name',
  }];
  const row = { name: 'testname' };
  const wrapper = shallowWithTheme(<TableRow columns={columns} row={row} keyField="id" index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableRow /> with columns should render correctly if a row has a keyField', () => {
  const columns = [{
    id: 123,
    name: 'Name',
    selector: 'name',
  }];
  const row = { id: 456, name: 'testname' };
  const wrapper = shallowWithTheme(<TableRow columns={columns} row={row} keyField="id" index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});
