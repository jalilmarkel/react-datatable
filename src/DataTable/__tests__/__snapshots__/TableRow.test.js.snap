// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with highlightOnHover prop 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`should render correctly with no columns 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
/>
`;

exports[`should render correctly with pointerOnHover prop 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`should render correctly with striped prop 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`when column.ignoreRowClick is true should not render the click clip div 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with children should render correctly 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with columns should render correctly if a row does not have a keyField 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with columns should render correctly if a row has a keyField 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c1:first-child {
  padding-left: calc(48px / 2);
}

.c1:last-child {
  padding-right: calc(48px / 2);
}

.c1 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <div
      class="c2"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with columns should render correctly if expandableRows 1`] = `
.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c3:first-child {
  padding-left: calc(48px / 2);
}

.c3:last-child {
  padding-right: calc(48px / 2);
}

.c3 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c2 {
  outline: none;
  border: none;
  display: block;
  width: 40px;
  height: 40px;
  background-color: transparent;
  background-image: url(data:image/svg+xml,%3Csvg%20fill%3D%22%23757575%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20width%3D%2224%22%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%3E%0A%20%20%20%20%3Cpath%20d%3D%22M8.59%2016.34l4.58-4.59-4.58-4.59L10%205.75l6%206-6%206z%22/%3E%0A%20%20%20%20%3Cpath%20d%3D%22M0-.25h24v24H0z%22%20fill%3D%22none%22/%3E%0A%3C/svg%3E);
  background-position: center center;
  background-repeat: no-repeat;
}

.c2:hover {
  cursor: pointer;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 56px;
  -ms-flex: 0 0 56px;
  flex: 0 0 56px;
  white-space: nowrap;
  font-weight: 400;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

.c1:not(:first-child) {
  padding-left: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="c1"
  >
    <button
      class="c2"
      data-testid="expander-button"
    />
  </div>
  <div
    class="rdt_TableCell c3"
  >
    <div
      class="c4"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with columns should render correctly if selectableRows is true 1`] = `
.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c2:first-child {
  padding-left: calc(48px / 2);
}

.c2:last-child {
  padding-right: calc(48px / 2);
}

.c2 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <input
      aria-label="select-row-456"
      name="select-row-456"
      style="font-size: 18px; cursor: pointer; margin-left: 9px;"
      type="checkbox"
    />
  </div>
  <div
    class="rdt_TableCell c2"
  >
    <div
      class="c3"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;

exports[`with columns should render correctly if selectableRows is true and a row is checked 1`] = `
.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c2:first-child {
  padding-left: calc(48px / 2);
}

.c2:last-child {
  padding-right: calc(48px / 2);
}

.c2 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  font-size: 13px;
  color: rgba(0,0,0,.87);
  min-height: 48px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
  background-color: transparent;
  color: rgba(0,0,0,.87);
}

<div
  class="rdt_TableRow c0"
>
  <div
    class="rdt_TableCell c1"
  >
    <input
      aria-label="select-row-456"
      checked=""
      name="select-row-456"
      style="font-size: 18px; cursor: pointer; margin-left: 9px;"
      type="checkbox"
    />
  </div>
  <div
    class="rdt_TableCell c2"
  >
    <div
      class="c3"
      data-tag="___react-data-table--click-clip___"
    />
    <div
      class="react-data-table--cell-content"
    >
      testname
    </div>
  </div>
</div>
`;
